### DISCUSSION TITLE
How to configure the script option "Skip executions triggered by PVs' first value" to avoid execution when reload displays?

### DISCUSSION THREAD

**Discussion by haitao127** [ControlSystemStudio/phoebus#1871] [Q&A] [answered]
When using CSS, there is an option within the script attached on a widget such as Action Button, if this option be checked, then the script will not execut when refresh the OPI.
![1](https://user-images.githubusercontent.com/12413287/114984857-b73dee80-9ec4-11eb-9d27-7f1cd6db4622.png)

But when using Phoebus, I can't find the option to set this configuration, so if I reload the display or reopen display/open the display for the first time, the script will execut once anyway, I think this is because when I do such operations, the PV is being connected/reconnected and triggered the script. But in many cases we don't want to trigger the script to execut if the PV is not changed, but I just want to reload the display for some other reasons such as some PV(will not trigger the script) on the display is disconnected...
![2](https://user-images.githubusercontent.com/12413287/114984880-bd33cf80-9ec4-11eb-9ce9-23e0421994f3.png)

Could anyone tell me how to implement this requirement in Phoebus? Thank you very much!

**Comment by kasemir** [comment, marked as answer]
In the display builder, a re-load is a complete shutdown and re-initialization start of a display.

In general, scripts should be avoided. A script should certainly not "do" anything. Scripts in displays need to be idempotent and only perform minor tasks related to the actual display, like transform received data from several PVs into a formatting suitable for a table. A script should never "do" something like write to PVs to perform automation or protection functionality.
If opening a display N times, including N instances of CS-Studio running in parallel or re-opening a display N times poses a problem, or if closing a display means you lose control system functionality because you depend on your script to "do" something to your control system, you are mis-using scripts and should put that functionality into a service (IOC, python with pcaspy, ...).

**Reply by haitao127** [reply]
Okay, thank you, we will reconsider the handling of scripts.
